docs/src/content/docs/components/using-components.mdx

```diff
@@ -0,0 +1,79 @@
+---
+title: Using components
+description: Using components in MDX and Markdoc with Starlight.
+sidebar:
+  order: 1
+---
+
+Components let you easily reuse a piece of UI or styling consistently.
+Examples might include a link card or a YouTube embed.
+Starlight supports the use of components in [MDX](https://mdxjs.com/) and [Markdoc](https://markdoc.dev/) files and provides some common components for you to use.
+
+[Learn more about building components in the Astro Docs](https://docs.astro.build/en/core-concepts/astro-components/).
+
+## Using a component in MDX
+
+You can use a component by importing it into your MDX file and then rendering it as a JSX tag.
+These look like HTML tags but start with an uppercase letter matching the name in your `import` statement:
+
+```mdx
+---
+# src/content/docs/example.mdx
+title: Welcome to my docs
+---
+
+import SomeComponent from '../../components/SomeComponent.astro';
+import AnotherComponent from '../../components/AnotherComponent.astro';
+
+<SomeComponent prop="something" />
+
+<AnotherComponent>
+	Components can also contain **nested content**.
+</AnotherComponent>
+```
```

Comment:
It would be nice to give some more concrete examples here (such as `<WrapperComponent />` for the last component). Thanks to https://www.rainsberger.ca/blog/50-43-whats-in-a-name/#_top for bringing this to mind.